src/app/space/app-launcher/services/app-launcher-gitprovider.service.ts

```diff
@@ -142,53 +134,31 @@ export class AppLauncherGitproviderService implements GitProviderService {
    * @returns {Observable<any>} list of existing GitHub repos
    */
   getGitHubRepoList(org: string): Observable<any> {
-    let url = this.END_POINT + this.API_BASE + 'repositories';
-    let location = org + '/';
-    if (this.gitHubUserLogin !== org) {
-      url += '?organization=' + org;
+    return this.getRepositories(org).pipe(map(AppLauncherGitproviderService.removeOrganizationPrefix));
+  }
+
+  private getRepositories(org: string = ''): Observable<string[]> {
+    if (org === this.gitHubUserLogin) {
+      org = '';
     }
-    return this.http
-      .get(url, { headers: this.headers }).pipe(
-      map((resp: HttpResponse<any>) => {
-          let repoList = [];
-          if (resp) {
-            let responseList: string[] = resp as any;
-            responseList.forEach(function(ele) {
-              repoList.push(ele.replace(location, ''));
-            });
-          }
-          return repoList;
-        }),
+    if (this.repositories[org]) {
+      return of(this.repositories[org]);
+    }
+    return this.http.get(this.createUrl(org), { headers: this.headers }).pipe(
+      map((json) => json ? json as string[] : []),
+      map((repositories) => this.repositories[org] = repositories),
```

Comment:
This `map` is only here to execute  side effects that should occur in a `tap` instead.